go-controller/pkg/ovn/egressgw.go

```diff
@@ -89,6 +92,9 @@ func (oc *DefaultNetworkController) addPodExternalGWForNamespace(namespace strin
 	for _, gwInfo := range nsInfo.routingExternalPodGWs {
 		existingGWs.Insert(gwInfo.gws.UnsortedList()...)
 	}
+	if config.OVNKubernetesFeature.EnableInterconnect && oc.zone != types.OvnDefaultZone {
+		existingGWs.Insert(nsInfo.routingExternalGWs.gws.UnsortedList()...)
+	}
```

Comment:
why is this conditionalized to single node zone IC? Isn't this something generic you are fixing?

Reply:
its conditionalized to multizone IC (not single zone).
because for nonIC and single zone IC we want to use the tranditional legacy method of annotation flush. The annotation k8s.ovn.org/external-gw-pod-ips must only contain ips of the nsInfo.routingExternalPodGWs NOT nsInfo.routingExternalGWs (this is already present as part of k8s.ovn.org/routing-external-gws so we don't want to duplicate it.) So TLDR; this condition is to compliment the 
```
} else {
		// flush here since we know we have added an egressgw pod and we also know the full list of existing gatewayIPs
```
below this block.